experiment-1-typical.ipynb

```diff
@@ -157,6 +143,18 @@
     "logging.basicConfig(level=logging.INFO)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set a socket timeout so that CF requests fail instead of hanging if they don't get a response\n",
```

Comment:
I don't recommend to limit PyWren to wait for CF invocations less than 60 seconds only. after recent PyWren's update there should not be any problems with that.

Reply:
I'd like to keep this in for my own sanity until we find the cause of my stall issues.

This timeout does not prevent long-running invocations. It only affects how long Python will wait for HTTPS responses to API calls, which should be very fast. The worst case I've seen (excluding stalls) is 13s for a response from the CF API, but normally it's less than 0.1s. I can successfully run the big dataset with a timeout of 2s.